src/pokemon_storage_system.c

```diff
@@ -5437,19 +5437,20 @@ static bool32 WaitForWallpaperGfxLoad(void)
 
 static void DrawWallpaper(const void *tilemap, s8 direction, u8 offset)
 {
-    s16 var = (offset * 2) + 3;
+    s16 var = offset * 256;
+    s16 var2 = (offset * 2) + 3;
     s16 x = ((sStorage->bg2_X / 8 + 10) + (direction * 24)) & 0x3F;
 
-    CopyRectToBgTilemapBufferRect(2, tilemap, 0, 0, 0x14, 0x12, x, 2, 0x14, 0x12, 0x11, offset << 8, var);
-
-    if (direction == 0)
-        return;
-    if (direction > 0)
-        x *= 1, x += 0x14; // x * 1 is needed to match, but can be safely removed as it makes no functional difference
-    else
-        x -= 4;
+    CopyRectToBgTilemapBufferRect(2, tilemap, 0, 0, 0x14, 0x12, x, 2, 0x14, 0x12, 0x11, var, var2);
 
+    if (direction)
+    {
+        if (direction > 0)
+            x += 0x14;
+        else
+            x -= 4;
     FillBgTilemapBufferRect(2, 0, x, 2, 4, 0x12, 0x11);
+    }
```

Comment:
The original early return would still match here, you only need to remove `x *= 1`. Regardless it has incorrect indentation

Reply:
oops, sec